prometheus_client/exposition.py

```diff
@@ -227,15 +235,23 @@ def sample_line(line):
     return ''.join(output).encode('utf-8')
 
 
-def choose_encoder(accept_header: str) -> Tuple[Callable[[CollectorRegistry], bytes], str]:
+def choose_formatter(accept_header: str) -> Tuple[Callable[[CollectorRegistry], bytes], str]:
     accept_header = accept_header or ''
     for accepted in accept_header.split(','):
-        if accepted.split(';')[0].strip() == 'application/openmetrics-text':
+        if accepted.split(';')[0].strip().lower() == 'application/openmetrics-text':
```

Comment:
Are these `.lower()` calls necessary?

Reply:
As this is the reference parser, it probably shouldn't accept incorrect input like this.

Reply:
for `accept-encoding`, the value is case insensitive according to the spec - https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.2.1
I (wrongly?) assumed this will be the case for the `accept` header as well, but checking it now, it is undefined in the spec - https://datatracker.ietf.org/doc/html/rfc7231#section-5.3.2
If you think we shouldn't add the `lower()` here for the `accept` headers, I am fine to remove this. However, I believe we should keep the `lower()` when checking the `accept-encoding` header values.